SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -0,0 +1,1833 @@
+		{
+		case EntityDelta::UPDATE:
+		{
+			for (const AuthorityChange& Change : Delta.AuthorityLostTemporarily)
+			{
+				AuthorityLost(Delta.EntityId, Change.ComponentSetId);
+			}
+			for (const AuthorityChange& Change : Delta.AuthorityLost)
+			{
+				AuthorityLost(Delta.EntityId, Change.ComponentSetId);
+			}
+			for (const ComponentChange& Change : Delta.ComponentsAdded)
+			{
+				ApplyComponentAdd(Delta.EntityId, Change.ComponentId, Change.Data);
+				ComponentAdded(Delta.EntityId, Change.ComponentId, Change.Data);
+			}
+			for (const ComponentChange& Change : Delta.ComponentUpdates)
+			{
+				ApplyComponentUpdate(Delta.EntityId, Change.ComponentId, Change.Update);
+				ComponentUpdated(Delta.EntityId, Change.ComponentId, Change.Update);
+			}
+			for (const ComponentChange& Change : Delta.ComponentsRefreshed)
+			{
+				ApplyComponentAdd(Delta.EntityId, Change.ComponentId, Change.CompleteUpdate.Data);
+				ComponentAdded(Delta.EntityId, Change.ComponentId, Change.CompleteUpdate.Data);
+			}
+			for (const ComponentChange& Change : Delta.ComponentsRemoved)
+			{
+				ComponentRemoved(Delta.EntityId, Change.ComponentId);
+			}
+			for (const AuthorityChange& Change : Delta.AuthorityGained)
+			{
+				AuthorityGained(Delta.EntityId, Change.ComponentSetId);
+			}
+			for (const AuthorityChange& Change : Delta.AuthorityLostTemporarily)
+			{
+				AuthorityGained(Delta.EntityId, Change.ComponentSetId);
+			}
+			break;
+		}
+		case EntityDelta::ADD:
+			PopulateDataStore(Delta.EntityId);
+			EntityAdded(Delta.EntityId);
+			break;
+		case EntityDelta::REMOVE:
+			EntityRemoved(Delta.EntityId);
+			ActorDataStore.Remove(Delta.EntityId);
+			break;
+		case EntityDelta::TEMPORARILY_REMOVED:
+			EntityRemoved(Delta.EntityId);
+			ActorDataStore.Remove(Delta.EntityId);
+			PopulateDataStore(Delta.EntityId);
+			EntityAdded(Delta.EntityId);
+			break;
+		default:
+			break;
+		}
+	}
+}
+
+UnrealMetadata* ActorSystem::GetUnrealMetadata(const Worker_EntityId EntityId)
+{
+	if (ActorDataStore.Contains(EntityId))
+	{
+		return &ActorDataStore[EntityId].Metadata;
+	}
+	return nullptr;
+}
+
+void ActorSystem::PopulateDataStore(const Worker_EntityId EntityId)
+{
+	ActorData& Components = ActorDataStore.Emplace(EntityId, ActorData{});
+	for (const ComponentData& Data : SubView->GetView()[EntityId].Components)
+	{
+		switch (Data.GetComponentId())
+		{
+		case SpatialConstants::SPAWN_DATA_COMPONENT_ID:
+			Components.Spawn = SpawnData(Data.GetUnderlying());
+			break;
+		case SpatialConstants::UNREAL_METADATA_COMPONENT_ID:
+			Components.Metadata = UnrealMetadata(Data.GetUnderlying());
+			break;
+		case SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID:
+			Components.OwningClientWorker = NetOwningClientWorker(Data.GetUnderlying());
+			break;
+		default:
+			break;
+		}
+	}
+}
+
+void ActorSystem::ApplyComponentAdd(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId, Schema_ComponentData* Data)
+{
+	switch (ComponentId)
```

Comment:
qq - where is the rest of the actor data stored?

Reply:
what's the rest of the actor data? generated components and things? the actor system only maintains the components it needs the deserialized from of to do its work, anything else is handled as before